lib/router.js

```diff
@@ -6,26 +6,46 @@ const mixins = hof.mixins;
 const i18nFuture = hof.i18n;
 const router = require('express').Router();
 const path = require('path');
+const fs = require('fs');
 
 module.exports = (config) => {
 
   const baseUrl = config.route.baseUrl || '/';
-  const routeFields = path.resolve(config.caller, config.route.fields || config.fields);
+  const name = config.route.name || baseUrl.replace('/', '');
+  const appPath = name ? `apps/${name}` : '.';
+  const paths = {
+    fields: config.fields ? path.resolve(config.caller, config.fields) : config.fields,
+    routeFields: path.resolve(config.caller, config.route.fields || `${appPath}/fields`),
+    templates: path.resolve(config.caller, config.route.views || `${appPath}/views`)
+  };
+
+  let fields = {};
+  let routeFields = {};
 
-  let fields;
+  try {
+    if (paths.fields) {
+      fields = require(paths.fields);
+    }
+  } catch (err) {
+    throw new Error('Cannot find fields');
+  }
 
   try {
-    fields = require(path.resolve(config.caller, config.fields));
+    routeFields = require(paths.routeFields);
   } catch (err) {
-    fields = err;
+    throw new Error('Cannot find route fields');
```

Comment:
Since we know the wrong path for route & fields, we can throw a more descriptive error:

```javascript
throw new Error(`Cannot find route fields "${paths.routeFields}"`);
```

same above for fields.